### PR TITLE
Sumproduct

### PR DESCRIPTION
**Pull request by naterush**
# Description

Adds SUMPRODUCT, for use in weighted average formulas.

# Testing

Add a few tests. Note the restricted types we allow here. I don't think there's a need for rolling ranges support in this function for now!

# Documentation

Naw.

### CODE REVIEW

**Comment by aarondr77** [comment]
I tried to recreate the test case and am getting all 0's. Its probably the same bug as yesterday?

Screenshot 2023-05-23 at 4 20 39 PM

**Comment by aarondr77** [comment]
This isn't matching Excel's behavior.

Screenshot 2023-05-23 at 4 24 04 PM

Screenshot 2023-05-23 at 4 24 16 PM

File: mitosheet/mitosheet/public/v3/sheet_functions/number_functions.py [hunk 1]
```diff
@@ -592,6 +592,54 @@ def SUM(*argv: Optional[NumberInputType]) -> NumberFunctionReturnType:
...
+        ]
+    }
+    """
+
+    # We need to make sure that all of the passed arguments are the same length
+    # so we can multiply them together. They must be all dataframes or all series
+    for arg in argv:
+        length = len(arg)
+        if length != len(argv[0]):
+            raise ValueError(f"SUMPRODUCT requires all arguments to be the same dimensions.")
```
Comment (aarondr77) [inline mitosheet/mitosheet/public/v3/sheet_functions/number_functions.py:625]: I'm not getting this error, instead I'm getting this: 

Screenshot 2023-05-23 at 4 25 30 PM

File: mitosheet/mitosheet/tests/public/v3/sheet_functions/number_functions/test_sumproduct.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import pytest
+import pandas as pd
+import numpy as np
+from mitosheet.public.v3.rolling_range import RollingRange
+
+from mitosheet.public.v3.sheet_functions.number_functions import SUMPRODUCT
+
+
+SUM_PRODUCT_VALID_TETS = [
```
Comment (aarondr77) [inline mitosheet/mitosheet/tests/public/v3/sheet_functions/number_functions/test_sumproduct.py:15]: TETS -> TESTS

File: mitosheet/mitosheet/tests/public/v3/sheet_functions/number_functions/test_sumproduct.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    ([pd.Series([1,2,3]), pd.Series([1,1,1])], 6),
+    ([pd.Series([1,2,3]), pd.Series([1,2,3])], 14),
+    ([pd.Series([1,2,3]), pd.Series([1,2,3.0])], 14.0),
+    ([pd.Series([1,2,3]), pd.Series([1,2,3]), pd.Series([1,2,3])], 1 + 8 + 27),
+    
+    # Dataframes and series
+    ([pd.DataFrame({'a': [1, 1, 1], 'b': [2, 2, 2]}), pd.DataFrame({'a': [1, 1, 1], 'b': [2, 2, 2]}), pd.DataFrame({'a': [1, 1, 1], 'b': [2, 2, 2]})], 3 + 8 * 3),
+]
+
+SUM_INVALID_CAST_TESTS = [
```
Comment (aarondr77) [inline mitosheet/mitosheet/tests/public/v3/sheet_functions/number_functions/test_sumproduct.py:26]: Can we add a test that makes sure rolling ranges does not work in the way we expect?

**Comment by naterush** [comment]
Merging when tests pass.

**Pull request merged**